tmt/steps/report/reportportal.py

```diff
@@ -0,0 +1,142 @@
+import dataclasses
+import io
+import os
+import types
+import zipfile
+from typing import List, Optional
+
+import click
+import simplejson.errors
+
+import tmt.steps.report
+import tmt.utils
+
+from . import junit
+
+junit_xml: Optional[types.ModuleType] = None
+
+
+def import_junit_xml() -> None:
+    """
+    Import junit_xml module only when needed
+
+    Until we have a separate package for each plugin.
+    """
+    global junit_xml
+    try:
+        import junit_xml
+    except ImportError:
+        raise tmt.utils.ReportError(
+            "Missing 'junit-xml', fixable by 'pip install tmt[report-junit]'.")
+
+
+@dataclasses.dataclass
+class ReportReportPortalData(tmt.steps.report.ReportStepData):
+    url: Optional[str] = None
+    token: Optional[str] = None
+    project: Optional[str] = None
+    launch_name: Optional[str] = None
+
+
+@tmt.steps.provides_method("reportportal")
+class ReportReportPortal(tmt.steps.report.ReportPlugin):
+    """
+    Report test results to a ReportPortal instance
+
+    Requires a TOKEN for authentication, a URL of the ReportPortal
+    instance and the PROJECT name. In addition to command line options
+    it's possible to use environment variables to set the url and token:
+
+        export TMT_REPORT_REPORTPORTAL_URL=...
+        export TMT_REPORT_REPORTPORTAL_TOKEN=...
+
+    The optional launch NAME is passed to ReportPortal. Assuming the URL
+    and TOKEN variables are provided by the environment, the config can
+    look like this:
+
+        report:
+            how: reportportal
+            project: baseosqe
+            launch-name: maven
+    """
+
+    _data_class = ReportReportPortalData
+
+    @classmethod
+    def options(cls, how: Optional[str] = None) -> List[tmt.options.ClickOptionDecoratorType]:
+        """ Prepare command line options """
+        return [
+            click.option(
+                "--url", envvar="TMT_REPORT_REPORTPORTAL_URL", metavar="URL", required=True,
+                help="The URL of the ReportPortal instance where the data should be sent to."),
+            click.option(
+                "--project", metavar="PROJECT", required=True,
+                help="The project name which is used to create the full URL."),
```

Comment:
Project can be specified in the plan config so we should not make it mandatory here.